src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -130,14 +134,41 @@
     <Field Name="IsNewInstance" Type="bool" Null="NotApplicable"/>
   </Node>
 
-  <!--
-  This node is used to represent an expression of a certain type, when attempting to bind its Index or Range pattern indexer
-  It does survive past initial binding but will be replaced with a synthesized argument in DAG lowering.
-  -->
+  <!-- This node is used to represent the argument of type Index or Range for an implicit Index or Range indexer. It does not survive lowering -->
   <Node Name="BoundIndexOrRangeIndexerPatternValuePlaceholder" Base="BoundValuePlaceholderBase">
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
   </Node>
 
+  <!-- This node is used to represent the receiver for an implicit Index or Range indexer. It does not survive lowering -->
+  <Node Name="BoundIndexOrRangeIndexerPatternReceiverPlaceholder" Base="BoundValuePlaceholderBase">
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <Field Name="ValEscape" Type="uint" Null="NotApplicable"/>
+    <!-- For CheckValue -->
+    <Field Name="Expression" Type="BoundExpression"/>
+  </Node>
+
+  <!-- This node represents the receiver for a list pattern. It does not survive lowering -->
+  <Node Name="BoundListPatternReceiverPlaceholder" Base="BoundValuePlaceholderBase">
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <Field Name="ValEscape" Type="uint" Null="NotApplicable"/>
+  </Node>
+
+  <!-- This node represents the Index value of accessing an element in a list pattern. It does not survive lowering -->
+  <Node Name="BoundListPatternUnloweredIndexPlaceholder" Base="BoundValuePlaceholderBase">
```

Comment:
>BoundListPatternUnloweredIndexPlaceholder [](http://example.com/codeflow?start=14&length=41)

Why do we need to say "Unlowered"? The comment says: "It does not survive lowering" Is that not sufficient?

Reply:
We're substituting it with an unlowered expression (which will be handled later during lowering) which is unusual.

Here's a snippet that illustrates the difference between this placeholder and most other placeholders:
```
        public override BoundNode? VisitListPatternUnloweredIndexPlaceholder(BoundListPatternUnloweredIndexPlaceholder node)
        {
            return Visit(PlaceholderReplacement(node));
        }

        public override BoundNode? VisitListPatternReceiverPlaceholder(BoundListPatternReceiverPlaceholder node)
        {
            return PlaceholderReplacement(node);
        }
```

Reply:
> We're substituting it with an unlowered expression (which will be handled later during lowering) which is unusual.

I thought in our offline conversation we agreed to not do that unusual thing any more. Let's sync up offline.